samples/correlation-csharp/FunctionAppCorrelation/SimpleCorrelationDemo.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using DurableTask.Core;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+
+namespace FunctionAppCorrelation
+{
+    public class SimpleCorrelationDemo
+    {
+        private readonly TelemetryClient telemetryClient;
+        private readonly HttpClient httpClient;
+
+        public SimpleCorrelationDemo(TelemetryConfiguration telemetryConfiguration, HttpClient client)
+        {
+            this.telemetryClient = new TelemetryClient(telemetryConfiguration);
+            this.httpClient = client;
+        }
+
+        [FunctionName(nameof(Orchestration_W3C))]
+        public async Task<List<string>> Orchestration_W3C(
+           [OrchestrationTrigger] IDurableOrchestrationContext context)
+        {
+            var correlationContext = CorrelationTraceContext.Current as W3CTraceContext;
```

Comment:
This would be a bit cleaner as the following:

```csharp
if (CorrelationTraceContext.Current is W3CTraceContext correlationContext)
{
    var trace = new TraceTelemetry(
        $"Activity Id: {correlationContext.TraceParent} ParentSpanId: {correlationContext.ParentSpanId}");
    trace.Context.Operation.Id = correlationContext.TelemetryContextOperationId;
    trace.Context.Operation.ParentId = correlationContext.TelemetryContextOperationParentId;
    this.telemetryClient.Track(trace);
}

Reply:
I actually don't know why that's clearer. Judging by the names of this demo, the correlation context is always meant to be a `W3CTraceContext` so I don't see why we should replace the `as` coercion to an `if`-check.

Reply:
I don't love null coalescing after an `as` conversion. It means that if for some reason the customer switched the CorrelationContext away from `W3CTraceContext` after we added support for a new protocol, then this would start failing in weird ways (specifically, a bunch of fields would silently go `null`.

I guess in that case, a preferable piece of code would look like this:

```csharp
if (!CorrelationTraceContext.Current is W3CTraceContext correlationContext)
{
    throw new InvalidOperationException($"This sample expects a correlation trace context of {nameof(W3CTraceContext)}, but the context isof type {CorrelationTraceContext.Current.GetType()}");
}

var trace = new TraceTelemetry(
    $"Activity Id: {correlationContext.TraceParent} ParentSpanId: {correlationContext.ParentSpanId}");
trace.Context.Operation.Id = correlationContext.TelemetryContextOperationId;
trace.Context.Operation.ParentId = correlationContext.TelemetryContextOperationParentId;
this.telemetryClient.Track(trace);
```

Reply:
Ah yes, I see what you mean. I wasn't thinking of the full implications of the coalescing operation. I'll go for your new suggestion. Thanks!